rxbindings/src/test/java/com/amplifyframework/rx/RxStorageBindingTest.java

```diff
@@ -208,7 +215,49 @@ public void uploadFileReturnsResult() throws InterruptedException {
         testObserver.await(TIMEOUT_MS, TimeUnit.MILLISECONDS);
         testObserver.assertValues(result);
         testProgressObserver.assertValueCount(5);
+    }
+
+    /**
+     * When {@link StorageCategoryBehavior#uploadInputStream(String, InputStream, Consumer, Consumer)} returns
+     * a {@link StorageUploadFileResult}, then the {@link Single} returned by
+     * {@link RxStorageCategoryBehavior#uploadInputStream(String, InputStream)} should emit that result.
+     * @throws InterruptedException Not expected.
+     */
+    @Test
+    public void uploadInputStreamReturnsResult() throws InterruptedException {
+        StorageUploadFileResult result = StorageUploadFileResult.fromKey(remoteKey);
+        doAnswer(invocation -> {
+            // 0 key, 1 local, 2 options, 3 onProgress, 4 onResult, 5 onError
+            final int indexOfResultConsumer = 4;
+            final int indexOfProgressConsumer = 3;
+            Consumer<StorageUploadFileResult> resultConsumer = invocation.getArgument(indexOfResultConsumer);
+            Consumer<StorageTransferProgress> progressConsumer = invocation.getArgument(indexOfProgressConsumer);
 
+            Observable.interval(100, TimeUnit.MILLISECONDS)
```

Comment:
We have an existing test that uses this mechanism, and it will sometimes fail. I think it is because the last progress update happens very close to the time of the result consumer.

I think it would be okay to not test the progress listener.